src/WorkItemMigrator/WorkItemImport/WitClient/WitClientUtils.cs

```diff
@@ -625,20 +627,44 @@ public void SaveWorkItemFields(WorkItem wi, Settings settings)
             {
                 throw new ArgumentException(nameof(wi));
             }
+            
+            if (wi.Fields.TryGetValue(WiFieldReference.BoardColumn, out var value))
+            {
+                wi.Fields.Remove(WiFieldReference.BoardColumn);  // the API refuses to update this field directly
+                var itemType = wi.Fields[WiFieldReference.WorkItemType]?.ToString();  // used for logging
+                var kanbanField = wi.Fields.Keys.FirstOrDefault(k => k.EndsWith("_Kanban.Column")); // kanban field is the real board-column field
+
+                if (wi.Rev == 0)
+                {
+                    Logger.Log(LogLevel.Warning, $"Work Item {wi.Id}, rev {wi.Rev} - BoardColumn can not be set to '{value}' because " +
+                                                 $"items with only one revision are not supported by this feature.");
+                }
+                else if (string.IsNullOrWhiteSpace(kanbanField))
+                {
+                    Logger.Log(LogLevel.Warning, $"Work Item {wi.Id}, rev {wi.Rev} - BoardColumn can not be set to '{value}' because " +
+                                                 $"items of type '{itemType}' are not supported by this feature.");
+                }
+                else
+                {
+                    Logger.Log(LogLevel.Debug, $"Work Item {wi.Id}, rev {wi.Rev} - Setting BoardColumn to '{value}'.");
+                    wi.Fields[kanbanField] = value;
+                }
+            }
 
             // Build json patch document from fields
             JsonPatchDocument patchDocument = new JsonPatchDocument();
             foreach (string key in wi.Fields.Keys)
             {
                 if (new string[] {
-                    WiFieldReference.BoardColumn,
                     WiFieldReference.BoardColumnDone,
                     WiFieldReference.BoardLane,
                 }.Contains(key))
+                {
+                    Logger.Log(LogLevel.Debug, $"Work Item {wi.Id} importing field {key} is not supported.");
```

Comment:
Evaluate whether this should be a Warning statement or not

Reply:
I don't have much of an opinion on this - it's really up to you - you could even remove this line altogether if you'd prefer.  This line previously used to also skip WiFieldReference.BoardColumn - and it does so silently.  Leaving users to wonder why their values didn't get migrated properly.
But this is a decision for the Product Owner, as far as I'm concerned :)

Reply:
Ok! Leave it as debug at them moment then.